xbmc/interfaces/generic/RunningScriptObserver.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (C) 2017-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "RunningScriptObserver.h"
+
+#include "interfaces/generic/ScriptInvocationManager.h"
+
+CRunningScriptObserver::CRunningScriptObserver(int scriptId, CEvent& evt)
+  : m_scriptId(scriptId),
+  m_event(evt),
+  m_thread(this, "ScriptObs"),
+  m_stopEvent(true)
+{
+  m_thread.Create();
+}
+
+void CRunningScriptObserver::Run()
+{
+  do
+  {
+    if (!CScriptInvocationManager::GetInstance().IsRunning(m_scriptId))
+    {
+      m_event.Set();
+      break;
+    }
+  } while (!m_stopEvent.WaitMSec(20));
```

Comment:
50 wakeups a second is a bit intense. why not wait between 100ms and 500ms?

Reply:
It's only a move of the existing behaviour. If you want to change that, probably best to split that change off in another commit

Reply:
Yup I only copied / moved this behaviour from `CPluginDirectory` to `CRunningScriptObserver`. I only want to refactor `CPluginDirectory` into usable classes and I don't want to change any behaviour.